adcc/ExcitedStates.py

```diff
@@ -194,17 +209,48 @@ def __init__(self, data, method=None, property_method=None):
         # Special stuff for special solvers
         if isinstance(data, EigenSolverStateBase):
             self.excitation_vectors = data.eigenvectors
-            self.excitation_energies = data.eigenvalues
+            self.__excitation_energies = data.eigenvalues
         else:
             if hasattr(data, "eigenvalues"):
-                self.excitation_energies = data.eigenvalues
+                self.__excitation_energies = data.eigenvalues
             if hasattr(data, "eigenvectors"):
                 self.excitation_vectors = data.eigenvectors
             if hasattr(data, "excitation_energies"):
-                self.excitation_energies = data.excitation_energies
+                self.__excitation_energies = data.excitation_energies
             if hasattr(data, "excitation_vectors"):
                 self.excitation_vectors = data.excitation_vectors
 
+        # Collect all excitation energy corrections
+        self.__excitation_energies_uncorrected = self.excitation_energies.copy()
+
+        if excitation_energy_corrections is None:
+            excitation_energy_corrections = {}
+        elif not isinstance(excitation_energy_corrections, dict):
+            raise TypeError("excitation_energy_corrections must be a dict.")
+
+        if hasattr(self.reference_state, "excitation_energy_corrections"):
+            excitation_energy_corrections.update(
+                self.reference_state.excitation_energy_corrections)
```

Comment:
Now comes the tricky business: I think what the user passes explicitly should take preference, therefore:
```
merged = self.reference_state.excitation_energy_corrections.copy()
merged.update(excitation_energy_corrections)
excitation_energy_corrections = merged
```

Reply:
Done.